src/blueprint.zig

```diff
@@ -372,6 +374,246 @@ pub const Pattern = struct {
 	}
 };
 
+const BlockLike = union(enum) {
+	block: Block,
+	blockType: u16,
+};
+
+fn generatePropertyEnum() type {
+	var tempFields: [@typeInfo(Block).@"struct".decls.len]std.builtin.Type.EnumField = undefined;
+	var count = 0;
+
+	for(std.meta.declarations(Block)) |decl| {
+		const declInfo = @typeInfo(@TypeOf(@field(Block, decl.name)));
+		if(declInfo != .@"fn") continue;
+		if(declInfo.@"fn".return_type != bool) continue;
+		if(declInfo.@"fn".params.len > 1) continue;
+
+		tempFields[count] = .{.name = decl.name, .value = count};
+		count += 1;
+	}
+
+	const outFields: [count]std.builtin.Type.EnumField = tempFields[0..count].*;
+
+	return @Type(.{.@"enum" = .{
+		.tag_type = u8,
+		.fields = &outFields,
+		.decls = &.{},
+		.is_exhaustive = true,
+	}});
+}
+
+pub const Mask = struct {
+	const AndList = ListUnmanaged(Entry);
+	const OrList = ListUnmanaged(AndList);
+
+	entries: OrList,
+
+	const or_ = '|';
+	const and_ = '&';
+	const inverse = '!';
+	const tag = '$';
+	const property = '@';
+
+	const Entry = struct {
+		inner: Inner,
+		isInverse: bool,
+
+		const Inner = union(enum) {
+			block: Block,
+			blockType: u16,
+			blockTag: Tag,
+			blockProperty: Property,
+
+			const Property = generatePropertyEnum();
+
+			fn initFromString(specifier: []const u8) !Inner {
+				switch(specifier[0]) {
+					tag => {
+						const blockTag = specifier[1..];
+						if(blockTag.len == 0) return error.TagNotFound;
```

Comment:
Already covered by the case below,
also technically it is a valid tag, so we should probably allow it if it exists.